persistence/relational-jdbc/src/main/java/org/apache/polaris/persistence/relational/jdbc/QueryGenerator.java

```diff
@@ -208,6 +209,59 @@ static QueryFragment generateWhereClause(
     return new QueryFragment(clause, parameters);
   }
 
+  @VisibleForTesting
+  static PreparedQuery generateVersionQuery() {
+    return new PreparedQuery("SELECT version_value FROM POLARIS_SCHEMA.VERSION", List.of());
+  }
+
+  /**
+   * Generate a SELECT query to find any entities that have a given realm & parent and that may with
+   * a given location. The check is performed without consideration for the scheme, so a path on one
+   * storage type may give a false positive for overlapping with another storage type. This should
+   * be combined with a check using `StorageLocation`.
+   *
+   * @param realmId A realm to search within
+   * @param parentId A parent entity to search within
+   * @param baseLocation The base location to look for overlap with, with or without a scheme
+   * @return The list of possibly overlapping entities that meet the criteria
+   */
+  @VisibleForTesting
+  public static PreparedQuery generateOverlapQuery(
+      String realmId, long parentId, String baseLocation) {
+    StorageLocation baseStorageLocation = StorageLocation.of(baseLocation);
+    String locationWithoutScheme = baseStorageLocation.withoutScheme();
+
+    List<String> conditions = new ArrayList<>();
+    List<Object> parameters = new ArrayList<>();
+
+    String[] components = locationWithoutScheme.split("/");
+    StringBuilder pathBuilder = new StringBuilder();
+
+    for (String component : components) {
+      pathBuilder.append(component).append("/");
+      conditions.add("location_without_scheme = ?");
+      parameters.add(pathBuilder.toString());
+    }
```

Comment:
[doubt] does it needs to opiniated based on depth of the directory ? a user recently pointed out this https://apache-polaris.slack.com/archives/C084XDM50CB/p1750346497095709 hence wanted to ask